src/frequenz/sdk/timeseries/_formula_engine/_formula_engine.py

```diff
@@ -189,20 +191,24 @@ async def apply(self) -> Sample[Quantity]:
     "HigherOrderFormulaBuilder3Phase",
 ]
 
-_GenericEngine = TypeVar("_GenericEngine", "FormulaEngine", "FormulaEngine3Phase")
+_GenericEngine = TypeVar(
+    "_GenericEngine",
+    "FormulaEngine",  # type: ignore
+    "FormulaEngine3Phase",  # type: ignore
```

Comment:
Why are these `type: ignore` needed?

Reply:
It is because of a python limitation.  Those types are generic,  but when putting them in a TypeVar,  we are unable to keep them generic.  For example, we can't say

```python
_GenericEngine = TypeVar(
    "_GenericEngine",
    "FormulaEngine[QuantityT]",
    "FormulaEngine3Phase[QuantityT]",
)
```
This also makes it slightly harder to use the `_GenericEngine` type.  But it is something.

Reply:
Same here.  Is the same file, so maybe just a global comment at the start of the file mentioning that some `type: ignore` were necessary because of X and Y issues is enough.